[enterprise-4.16] TELCODOCS-1861: Add Lifecycle Agent Enhancements 4.17 (#86758)
openshift/openshift-docs: modules/ztp-image-based-upgrade-prep-oadp.adoc

* Deploy a version of {lcao} that is compatible with the version used with the seed image.
* Install the {oadp-short} Operator, the `DataProtectionApplication` CR, and its secret on the target cluster.
* Create an S3-compatible storage solution and a ready-to-use bucket with proper credentials configured. For more information, see "Installing and configuring the {oadp-short} Operator with {ztp}".

> * The `openshift-adp` namespace for the OADP `ConfigMap` object must exist on all managed clusters and the hub for the OADP `ConfigMap` to be generated and copied to the clusters.

Reviewer: ```suggestion
* The `openshift-adp` namespace for the OADP `ConfigMap` object must exist on all managed clusters and the hub for the OADP `ConfigMap`, so that X can generate the namespace and copy it to the required clusters.
```

====
--

. Create a `kustomization.yaml` with the following content:
+
[source,yaml]
----
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

configMapGenerator: # <1>
- files:
  - source-crs/ibu/PlatformBackupRestoreWithIBGU.yaml
  #- source-crs/custom-crs/ApplicationClusterScopedBackupRestore.yaml
  #- source-crs/custom-crs/ApplicationApplicationBackupRestoreLso.yaml
  name: oadp-cm
  namespace: openshift-adp # <2>
generatorOptions:
  disableNameSuffixHash: true
----
<1> Creates the `oadp-cm` `ConfigMap` object on the hub cluster with `Backup` and `Restore` CRs.

> <2> The namespace must exist on all managed clusters and the hub for the OADP `ConfigMap` to be generated and copied to the clusters.

Reviewer: See a previous comment on this similar sentence.
